### PR TITLE
Fix test

### PR DESCRIPTION
**Pull request by allevo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/consumer.js [hunk 1]
```diff
@@ -4,9 +4,13 @@ const EE = require('events').EventEmitter
...
 
-function Consumer (opts, log, next) {
+function Consumer (opts, log, next, topicConf) {
   this._log = log
-  this.consumer = new Kafka.KafkaConsumer(opts)
+  if (topicConf) {
+    this.consumer = new Kafka.KafkaConsumer(opts, topicConf)
+  } else {
+    this.consumer = new Kafka.KafkaConsumer(opts)
+  }
```
Comment (mcollina) [inline lib/consumer.js]: Why this change? Could we just initialize topicConf to an empty object.

File: lib/producer.js [hunk 2]
```diff
@@ -4,9 +4,13 @@ const EE = require('events').EventEmitter
...
 
-function Consumer (opts, log, next) {
+function Consumer (opts, log, next, topicConf) {
   this._log = log
-  this.producer = new Kafka.Producer(opts)
+  if (topicConf) {
+    this.producer = new Kafka.Producer(opts, topicConf || {})
+  } else {
+    this.producer = new Kafka.Producer(opts)
+  }
```
Comment (mcollina) [inline lib/producer.js]: Same

File: test.js [hunk 3]
```diff
@@ -6,100 +6,146 @@ const crypto = require('crypto')
...
-        fastify.kafka.push({
-          topic: 'test',
+        producerFastify.kafka.push({
+          topic: topicName,
           payload: 'hello world!',
           key: 'testKey'
         })
+
+        t.ok(true)
       }, 1000)
```
Comment (mcollina) [inline test.js]: while you are refactoring, can you remove this timeout?

File: test.js [hunk 4]
```diff
@@ -6,100 +6,146 @@ const crypto = require('crypto')
...
     })
 })
+
+function generateGroupId () {
+  return crypto.randomBytes(20).toString('hex')
+}
+
+function copyPlainObject (obj) {
+  return JSON.parse(JSON.stringify(obj))
+}
```
Comment (mcollina) [inline test.js]: Can you add a comment that states it is really SLOW and to not use it outside of tests.

File: lib/consumer.js (hunk 1 above)
Comment (allevo) [reply lib/consumer.js]: done

File: lib/producer.js (hunk 2 above)
Comment (allevo) [reply lib/producer.js]: done

File: test.js (hunk 4 above)
Comment (allevo) [reply test.js]: done

File: test.js (hunk 3 above)
Comment (allevo) [reply test.js]: done

**Comment by allevo** [comment]
For my purpose I have to update `fastify` too.
Al least to 1.
Can I use this PR for that?

File: index.js [hunk 5]
```diff
@@ -42,4 +42,4 @@ function buildConsumer (fastify, opts, next) {
   }
 }
 
-module.exports = fp(fastifyKafka, '>=0.30.0')
+module.exports = fp(fastifyKafka, '>=1.0.0')
```
Comment (mcollina) [inline index.js]: make it two!

Comment (allevo) [reply index.js]: done

**Comment by allevo** [comment]
In last commit there's the implementation for the `consume` callback.
See https://github.com/Blizzard/node-rdkafka#standard-api-1

File: .travis.yml [hunk 6]
```diff
@@ -1,10 +1,9 @@
 language: node_js
 
 node_js:
-  - "9"
+  - "12"
+  - "10"
   - "8"
```
Comment (delvedor) [inline .travis.yml]: I think we can drop Node v8 as well.

Comment (allevo) [reply .travis.yml]: done

**Comment by mcollina** [comment]
@allevo would you mind merging this and issuing a new release? You should have publish rights

**Comment by allevo** [comment]
@mcollina I have no rights for merging to PR.
But I can publish it.

**Comment by mcollina** [comment]
Now you can!

**Comment by allevo** [comment]
Published 0.2.0
